File: dev-docs/snippets/events/workflow-started.swift
import JsBaoClient

// Fires when a workflow run starts.
func workflowStarted(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.workflowStarted) { (event: WorkflowStartedEvent) in
    // Swift carries only workflowKey + runId; JS also has workflowId/runKey/instanceId/contextDocId/meta.
    print("started", event.workflowKey, event.runId)
  }
  // #endregion example
  _ = sub
}

File: dev-docs/snippets/events/workflow-started.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires when a workflow run starts.
export function workflowStarted(client: JsBaoClient) {
  // #region example
  client.on("workflowStarted", (payload) => {
    console.log("started", payload.workflowKey, payload.runId, payload.instanceId);
  });
  // #endregion example
}
